control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -423,165 +423,153 @@ func SetupGatewayControllerWithManager(ctx context.Context, mgr ctrl.Manager, co
 		Owns(&corev1.Service{}).
 		Owns(&corev1.Pod{}).
 		Watches(
-			source.NewKindWithCache(&gwv1beta1.ReferenceGrant{}, mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformReferenceGrant(ctx)),
+			&gwv1beta1.ReferenceGrant{},
+			handler.EnqueueRequestsFromMapFunc(r.transformReferenceGrant),
 		).
 		Watches(
-			source.NewKindWithCache(&gwv1beta1.GatewayClass{}, mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformGatewayClass(ctx)),
+			&gwv1beta1.GatewayClass{},
+			handler.EnqueueRequestsFromMapFunc(r.transformGatewayClass),
 		).
 		Watches(
-			source.NewKindWithCache(&gwv1beta1.HTTPRoute{}, mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformHTTPRoute(ctx)),
+			&gwv1beta1.HTTPRoute{},
+			handler.EnqueueRequestsFromMapFunc(r.transformHTTPRoute),
 		).
 		Watches(
-			source.NewKindWithCache(&gwv1alpha2.TCPRoute{}, mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformTCPRoute(ctx)),
+			&gwv1alpha2.TCPRoute{},
+			handler.EnqueueRequestsFromMapFunc(r.transformTCPRoute),
 		).
 		Watches(
-			source.NewKindWithCache(&corev1.Secret{}, mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformSecret(ctx)),
+			&corev1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(r.transformSecret),
 		).
 		Watches(
-			source.NewKindWithCache(&v1alpha1.MeshService{}, mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformMeshService(ctx)),
+			&v1alpha1.MeshService{},
+			handler.EnqueueRequestsFromMapFunc(r.transformMeshService),
 		).
 		Watches(
-			source.NewKindWithCache(&corev1.Endpoints{}, mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformEndpoints(ctx)),
+			&corev1.Endpoints{},
+			handler.EnqueueRequestsFromMapFunc(r.transformEndpoints),
 		).
 		Watches(
-			&source.Kind{Type: &corev1.Pod{}},
-			handler.EnqueueRequestsFromMapFunc(r.transformPods(ctx)),
+			&corev1.Pod{},
+			handler.EnqueueRequestsFromMapFunc(r.transformPods),
 			builder.WithPredicates(predicate),
 		).
-		Watches(
+		WatchesRawSource(
```

Comment:
Need to use `WatchesRawSource` now when watching a channel instead of just a kind